airflow/models/taskinstance.py

```diff
@@ -584,8 +584,12 @@ def __init__(
         # can be changed when calling 'run'
         self.test_mode = False
 
-        self.dataset_event_manager = conf.getimport(
-            'core', 'dataset_event_manager_class', fallback='airflow.datasets.manager.DatasetEventManager'
+    @cached_property
+    def dataset_event_manager(self):
+        return conf.getimport(
+            section='core',
+            key='dataset_event_manager_class',
+            fallback='airflow.datasets.manager.DatasetEventManager',
         )()
```

Comment:
Why does this need to be an instance-wide property in the first place? Do we plan to add hook so each task instance can have a different manager?

Reply:
i'm actually working on centralizing this logic... it won't just be used on on TIs @uranusjr @blag